plugins/outputs/nebius_cloud_monitoring/nebius_cloud_monitoring_test.go

```diff
@@ -0,0 +1,138 @@
+				require.NoError(t, err)
+			} else if strings.HasSuffix(r.URL.Path, "/folder") {
+				_, err := io.WriteString(w, "folder1")
+				require.NoError(t, err)
+			}
+			w.WriteHeader(http.StatusOK)
+		}),
+	)
+	defer testMetadataHTTPServer.Close()
+	metadataTokenURL := "http://" + testMetadataHTTPServer.Listener.Addr().String() + "/token"
+	metadataFolderURL := "http://" + testMetadataHTTPServer.Listener.Addr().String() + "/folder"
+
+	ts := httptest.NewServer(http.NotFoundHandler())
+	defer ts.Close()
+	url := "http://" + ts.Listener.Addr().String() + "/metrics"
+
+	tests := []struct {
+		name    string
+		plugin  *NebiusCloudMonitoring
+		metrics []telegraf.Metric
+		handler func(t *testing.T, w http.ResponseWriter, r *http.Request)
+	}{
+		{
+			name:   "metric is converted to json value",
+			plugin: &NebiusCloudMonitoring{},
+			metrics: []telegraf.Metric{
+				testutil.MustMetric(
+					"cluster",
+					map[string]string{},
+					map[string]interface{}{
+						"cpu": 42.0,
+					},
+					time.Unix(0, 0),
+				),
+			},
+			handler: func(t *testing.T, w http.ResponseWriter, r *http.Request) {
+				message := readBody(r)
+				require.Len(t, message.Metrics, 1)
+				require.Equal(t, "cluster_cpu", message.Metrics[0].Name)
+				require.Equal(t, 42.0, message.Metrics[0].Value)
+				w.WriteHeader(http.StatusOK)
+			},
+		},
+		{
+			name:   "int64 metric is converted to json value",
+			plugin: &NebiusCloudMonitoring{},
+			metrics: []telegraf.Metric{
+				testutil.MustMetric(
+					"cluster",
+					map[string]string{},
+					map[string]interface{}{
+						"value": int64(9223372036854775806),
+					},
+					time.Unix(0, 0),
+				),
+			},
+			handler: func(t *testing.T, w http.ResponseWriter, r *http.Request) {
+				message := readBody(r)
+				require.Len(t, message.Metrics, 1)
+				require.Equal(t, "cluster_value", message.Metrics[0].Name)
+				require.Equal(t, float64(9.223372036854776e+18), message.Metrics[0].Value)
+				w.WriteHeader(http.StatusOK)
+			},
+		},
+		{
+			name:   "int metric is converted to json value",
+			plugin: &NebiusCloudMonitoring{},
+			metrics: []telegraf.Metric{
+				testutil.MustMetric(
+					"cluster",
+					map[string]string{},
+					map[string]interface{}{
+						"value": 9226,
+					},
+					time.Unix(0, 0),
+				),
+			},
+			handler: func(t *testing.T, w http.ResponseWriter, r *http.Request) {
+				message := readBody(r)
+				require.Len(t, message.Metrics, 1)
+				require.Equal(t, "cluster_value", message.Metrics[0].Name)
+				require.Equal(t, float64(9226), message.Metrics[0].Value)
+				w.WriteHeader(http.StatusOK)
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ts.Config.Handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				tt.handler(t, w, r)
+			})
+			tt.plugin.Log = testutil.Logger{}
+			tt.plugin.EndpointURL = url
+			tt.plugin.MetadataTokenURL = metadataTokenURL
+			tt.plugin.MetadataFolderURL = metadataFolderURL
```

Comment:
How about having a local test instance?
```suggestion
			plugin := &NebiusCloudMonitoring{
				EndpointURL:       url,
				MetadataTokenURL:  metadataTokenURL,
				MetadataFolderURL: metadataFolderURL,
				Log:               testutil.Logger{},
			}			
```